### PR TITLE
bump controller-runtime version

### PR DESCRIPTION
**Pull request by pxsalehi**
**Description**

Changes proposed in this pull request:

- Update the controller-runtime dependency

**Related issue(s)**

### CODE REVIEW

**Review by themue — APPROVED** [review]

File: components/eventing-controller/config/crd/bases/eventing.kyma-project.io_eventingbackends.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.4.0
+    controller-gen.kubebuilder.io/version: v0.2.5
```
Comment (sayanh) [inline components/eventing-controller/config/crd/bases/eventing.kyma-project.io_eventingbackends.yaml:7]: Let's upgrade to https://github.com/kubernetes-sigs/controller-tools/releases/tag/v0.6.0 if you are willing to.

Then you need to change the [Makefile](https://github.com/kyma-project/kyma/blob/main/components/eventing-controller/Makefile#L136) as well.

Finally, execute the `make copy-crds` target to sync CRDs in the resources folder.

File: components/eventing-controller/reconciler/backend/reconciler_test.go [hunk 2]
```diff
@@ -416,8 +416,8 @@ var _ = Describe("Backend Reconciliation Tests", func() {
 					EventingReady:               utils.BoolPtr(false),
 					SubscriptionControllerReady: utils.BoolPtr(false),
 					PublisherProxyReady:         utils.BoolPtr(false),
-					BebSecretName:               "",
-					BebSecretNamespace:          "",
+					BebSecretName:               bebSecret1name,
```
Comment (sayanh) [inline components/eventing-controller/reconciler/backend/reconciler_test.go:419]: There should not be any bebsecret in the status as reconciler failed to stop the NATS controller. I think empty should be the correct value.

Comment (pxsalehi) [reply components/eventing-controller/reconciler/backend/reconciler_test.go:419]: We are using BEB as backend and the name/namespace is what is expected. Not ready status is due to not stopping the nats subscription controller. I think this also makes sense.

Comment (sayanh) [reply components/eventing-controller/reconciler/backend/reconciler_test.go:419]: Right, I overlooked, bebsecret name/namespace do get updated.

**Review by sayanh — DISMISSED** [review]

**Review by sayanh — APPROVED** [review]

**Pull request merged**